dnf/lock.py

```diff
@@ -64,6 +64,15 @@ def __init__(self, target, description, blocking=False):
         self.target = target
         self.thread_lock = threading.RLock()
 
+    def _try_ensure_dir(self):
```

Comment:
So, since ```os.makedirs(..., exist_ok=True)``` does not exist in Python 2, we should probably rewrite ```ensure_dir``` to be similar to how ```os.makedirs(..., exist_ok=True)``` works: https://hg.python.org/cpython/file/cabd257b1070/Lib/os.py#l241
So, something like:
```
try:
    os.makedirs(dname)
except OSError:
    if e.errno != errno.EEXIST or not path.isdir(name):
        raise
```
It will fix the problem too, I think...